File: frontend/src/api/models/UnitCreateRequestDTO.ts
import type { AddressDTO } from './AddressDTO.ts'

export type UnitCreateRequestDTO = {
  /**
   * @type string
   */
  block: string
  /**
   * @type string
   */
  lot: string
  /**
   * @type object
   */
  address: AddressDTO
}

File: backend/src/main/java/br/com/visitsafe/dto/unit/UnitCreateRequestDTO.java
package br.com.visitsafe.dto.unit;

import br.com.visitsafe.dto.AddressDTO;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UnitCreateRequestDTO {

    @NotBlank
    private String block;

    @NotBlank
    private String lot;

    @NotNull
    @Valid
    private AddressDTO address;
}
